xbmc/cores/VideoPlayer/VideoRenderers/HwDecRender/RendererMediaCodecSurface.cpp

```diff
@@ -156,6 +156,17 @@ bool CRendererMediaCodecSurface::RenderUpdateVideoHook(bool clear, DWORD flags,
     mci->RenderUpdate(srcRect, dstRect);
   }
 
+  /** @todo This is a really bad hack and not meant to be
+  * a solution. This needs proper fixing when android maintainer
+  * is back. Real solution: Block until your next vblank.
```

Comment:
> Bottom-line: What are bypass renderers supposed to wait for?

Depending on how accurately you want to sync with audio and subtitles, you can

1) use this hack, results in poor sync and maybe skipped frames when subs are involved
2) throttle render loop to fps by using time only. This is rather trivial. All you need is a member that remembers last rendering times. Sync may be off because player has no info when a particular frame was rendered.
3) Proper sync by lettering player know when a particular frame was rendered to display.